lua/octo/init.lua

```diff
@@ -772,32 +772,82 @@ function M.show_summary()
     return
   end
 
+  local repo, number
   local cursor = api.nvim_win_get_cursor(0)
   local current_line = vim.fn.getline(".")
   local different_repo_issue_pattern = "%s([^ /]+/[^ #]+)#(%d+)%s"
   local same_repo_issue_pattern = "%s#(%d+)%s"
-  local start_col, end_col, repo, number = current_line:find(different_repo_issue_pattern)
-
-  if start_col and end_col and start_col <= cursor[2] and cursor[2] <= end_col then
-    local popup_bufnr = api.nvim_create_buf(false, true)
-    api.nvim_buf_set_lines(popup_bufnr, 0, -1, false, {repo, number})
-    window.create_popup({
-      bufnr = popup_bufnr,
-      width = 40,
-      height = 10
-    })
+  local start_col1, end_col1, repo1, number1 = current_line:find(different_repo_issue_pattern)
+
+  if start_col1 and end_col1 and start_col1 <= cursor[2] and cursor[2] <= end_col1 then
+    repo = repo1
+    number = number1
   else
     local start_col2, end_col2, number2 = current_line:find(same_repo_issue_pattern)
     if start_col2 and end_col2 and start_col2 <= cursor[2] and cursor[2] <= end_col2 then
-      local popup_bufnr2 = api.nvim_create_buf(false, true)
-      api.nvim_buf_set_lines(popup_bufnr2, 0, -1, false, {repo2, number2})
-      window.create_popup({
-        bufnr = popup_bufnr2,
-        width = 40,
-        height = 10
-      })
+      repo = repo2
+      number = number2
     end
   end
+  if not repo or not number then
+    return
+  end
+  local owner = vim.split(repo, "/")[1]
```

Comment:
This function is waaaay too long. It doesn't fit on a single screen (bad measurement, but a good old one). If I want to get a grasp of how the preview stuff works, I have to fight through all the details from the very top to the bottom. Usually you have forgotten where you came from while you are stuck in the low level details of the function. I would recommend to split it into multiple parts. Another good indicator for if a function is too complex: could you simply imagine the test cases it would need to write for this interface function (assuming you don't use a TDD workflow)?

Reply:
Ok, I think I improved enough without splitting it, but may need to do it if I end up adding hovering over user names or other objects